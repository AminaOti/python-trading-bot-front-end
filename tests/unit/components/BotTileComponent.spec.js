import botTileComponent from "@/components/bodyComponents/BotTileComponent.vue";
import { shallowMount } from "@vue/test-utils";
const MOCK_BOT_NAME = "Mock bot name";
const MOCK_BOT_LOGO = "images/testImage.jpeg";

const shallowMountcomponentBotTileComponent = (props) =>
  shallowMount(botTileComponent, {
    propsData: props || {
      botName: MOCK_BOT_NAME,
      botLogo: MOCK_BOT_LOGO,
    },
  });

describe("Bot tile (child component) - BotTileComponent.vue", () => {
  describe("should receive the title and logo image from a parent component", () => {
    it("should receive a bot name from the parent component (props)", () => {
      const component = shallowMountcomponentBotTileComponent();
      expect(component.props().botName).toBe(MOCK_BOT_NAME);
    });
    it("should receive a logo image from the parent component (props)", () => {
      const component = shallowMountcomponentBotTileComponent();
      expect(component.props().botLogo).toBe(MOCK_BOT_LOGO);
    });
  });
});
