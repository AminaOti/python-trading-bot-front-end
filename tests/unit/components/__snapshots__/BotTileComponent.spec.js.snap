// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bot tile (child component) - BotTileComponent.vue should match snapshot 1`] = `
<div
  class="bot-card"
  data-test="bot-tile"
>
  <div
    class="grid-container"
  >
    <div
      class="grid-child"
    >
      <img
        alt="thicc-dough-crypto-logo"
        class="grid-child-left-column"
        src=""
      />
    </div>
    <div
      class="grid-child-right-column"
    >
      <p>
        Mock bot name
      </p>
    </div>
  </div>
</div>
`;
